src/types.ts

```diff
@@ -11,7 +11,7 @@ export interface Connection {
 	info?: <T extends Record<string, unknown> = Record<string, unknown>>() =>
 		Promise<T | undefined>;
 
-	signup: (vars: ScopeAuth) => Promise<Token>;
+	signup: (vars: Partial<ScopeAuth> & Pick<ScopeAuth, "SC">) => Promise<Token>;
```

Comment:
Instead of doing this picking and partially-ing, let's perhaps make the NS and DB props optional in the Auth types themselves? Should make things a bit more readable

Reply:
That's way better. I didn't want to start messing with your types right away.